### PR TITLE
open-mesh: Add rpath to cmake build.  Add rpath to test linking to ma…

### PR DESCRIPTION
**Pull request by areese**
…ke open-mesh build on Apple Silicon

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/open-mesh.rb [hunk 1]
```diff
@@ -67,6 +67,7 @@ def install
       -lOpenMeshCore
       -lOpenMeshTools
       --std=c++11
+      -Wl,-rpath,#{lib}
```
Comment (carlocab) [inline Formula/open-mesh.rb:70]: This one's a bit concerning. What's the error you get when you don't add this?

Comment (carlocab) [reply Formula/open-mesh.rb:70]: Hm, I think I see. `libOpenMeshTools` has linkage with an `@rpath/libOpenMeshCore.8.1.dylib`. It's a bit sad the dynamic linker isn't able to find it when the `test` executable has explicit linkage with `libOpenMeshCore`.

Comment (areese) [reply Formula/open-mesh.rb:70]: yes that's what it was.
I made these patches a while ago, so any questions I have to go test them out again.

Comment (areese) [reply Formula/open-mesh.rb:70]:
```
% brew test open-mesh                       
==> Testing open-mesh
==> /usr/bin/clang++ test.cpp -o test -I/opt/homebrew/Cellar/open-mesh/8.1/include -L/opt/homebrew/Cellar/open-mesh/8.1/lib -lOpenMeshCore -lOpenMeshTools --std=c++11
==> ./test
Last 15 lines from Library/Logs/Homebrew/open-mesh/test.02.test:
2021-06-29 16:26:08 -0700

./test

dyld: Library not loaded: @rpath/libOpenMeshCore.8.1.dylib
  Referenced from: /opt/homebrew/opt/open-mesh/lib/libOpenMeshTools.8.1.dylib
  Reason: image not found
Error: open-mesh: failed
An exception occurred within a child process:
  BuildError: Failed executing: ./test
/opt/homebrew/Library/Homebrew/formula.rb:2218:in `block in system'
/opt/homebrew/Library/Homebrew/formula.rb:2154:in `open'
/opt/homebrew/Library/Homebrew/formula.rb:2154:in `system'
/opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/open-mesh.rb:72:in `block in <class:OpenMesh>'
/opt/homebrew/Library/Homebrew/formula.rb:2018:in `block (3 levels) in run_test'
/opt/homebrew/Library/Homebrew/utils.rb:550:in `with_env'
/opt/homebrew/Library/Homebrew/formula.rb:2017:in `block (2 levels) in run_test'
/opt/homebrew/Library/Homebrew/formula.rb:925:in `with_logging'
/opt/homebrew/Library/Homebrew/formula.rb:2016:in `block in run_test'
/opt/homebrew/Library/Homebrew/mktemp.rb:63:in `block in run'
/opt/homebrew/Library/Homebrew/mktemp.rb:63:in `chdir'
/opt/homebrew/Library/Homebrew/mktemp.rb:63:in `run'
/opt/homebrew/Library/Homebrew/formula.rb:2269:in `mktemp'
/opt/homebrew/Library/Homebrew/formula.rb:2010:in `run_test'
/opt/homebrew/Library/Homebrew/test.rb:43:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:93:in `block in timeout'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `block in catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:108:in `timeout'
/opt/homebrew/Library/Homebrew/test.rb:48:in `<main>'
```

**Review by carlocab — APPROVED** [review]

**Pull request merged**